sdk/core/azure_core_amqp/src/management.rs

```diff
@@ -24,6 +24,105 @@ pub trait AmqpManagementApis {
         application_properties: AmqpOrderedMap<String, AmqpValue>,
     ) -> impl std::future::Future<Output = Result<AmqpOrderedMap<String, AmqpValue>>>;
 }
+pub(crate) mod error {
+    use std::{error::Error, fmt::Debug};
+
+    use crate::{
+        error::{AmqpErrorKind, AmqpReceiverError, AmqpSenderError},
+        AmqpError,
+    };
+
+    pub enum AmqpManagementError {
+        AmqpManagementAlreadyAttached,
+        AmqpManagementNotAttached,
+
+        /// An error has occurred with Sending the management request.
+        SendError(AmqpSenderError),
+        ReceiveError(AmqpReceiverError),
+
+        InvalidManagementResponse(String),
+        DecodingError,
+        NotAccepted,
+        Disposition,
+        HttpStatusCode(azure_core::StatusCode, Option<String>),
+    }
+
+    impl std::fmt::Display for AmqpManagementError {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            match &self {
+                AmqpManagementError::AmqpManagementAlreadyAttached => {
+                    f.write_str("AMQP Management is already attached")
+                }
+                AmqpManagementError::AmqpManagementNotAttached => {
+                    f.write_str("AMQP Management is not attached")
+                }
+                AmqpManagementError::InvalidManagementResponse(s) => {
+                    if let Some(e) = self.source() {
+                        f.write_fmt(format_args!("Invalid Management Response: {}: {}", s, e))
```

Comment:
Tip: as long as you don't need an expression - just a simpel var reference - you can do:
```suggestion
                        f.write_fmt(format_args!("Invalid Management Response: {s}: {e}"))
```
IMO, cleaner, but either is fine.

Reply:
Yeah, habits die hard (too much time in .Net :)).